articles/migrations/guides/extensibility-node12.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "Migration Guide: Extensibility and Node 12"
+description: Learn about the Auth0 features affected by the Node.js v8 to Node.js v12 migration and review our recommendations for ensuring a smooth migration process.
+toc: true
+topics:
+  - migrations
+  - extensibility
+  - nodejs
+  - rules
+  - hooks
+  - custom-db
+  - custom-social-connections
+  - extensions
+contentType:
+  - concept
+  - how-to
+useCase:
+  - manage-accounts
+  - migrate
+---
+# Migration Guide: Extensibility and Node 12
+
+On December 31, 2019, [Node.js v8 went out of long-term support (LTS)](https://github.com/nodejs/Release#release-schedule), which means that the Node.js development team no longer back-ports critical security fixes to this version. This _could_ expose your extensibility code to security vulnerabilities.
+
+As such, Auth0 is migrating from Node 8 to Node 12.
+
+**We will be ending support for the Node 8 runtime on June xx, 2020**. Tenants which have not already migrated to Node 12 will be migrated automatically. Automatic migrations will begin on April xx, 2020, and continue through June xx, 2020. Automatic migrations are happening in cohorts: 
+
+* Free tier tenants will be migrated beginning on April xx, 2020
+* Self-serve tenants will follow through early May 2020
+* Enterprise tenants will be migrated in late May and June 2020
+
+Customers will be notified two weeks in advance of their automatic migration date, with additional periodic reminders leading up to their automatic migration date.
+
+::: warning
+All tenant automatic upgrades to Node 12 will be completed no later than June xx, 2020. Most tenant upgrades will be completed before this date. Admins for affected tenants will be notified of their exact migration date.
+:::
+
+In this document, we:
+
+* Provide recommendations on how you can ensure a smooth migration for your environment
+* Detail the specific modules affected
+
+## Summary of the migration
+
+The Webtask runtime powering the following Auth0 features utilize Node 8:
+
+* Rules
+* Hooks
+* Custom database connections
+* Custom social connections
+* Extensions
```

Comment:
There is one section regarding extensions that we would like to add to this document.

We are asking other teams to verify that their extensions work properly in Node 12. As they do, we'd like to add them to a list of "blessed" Extensions for Node 12 so that customers have a clear understanding of which ones are supported in Node 12.

As of this time, we have verified:

- Realtime Webtask Logs
- Deploy extensions (Github, Gitlab, etc.)
- Deploy CLI